script/check-patch-diff.js

```diff
@@ -0,0 +1,18 @@
+const { spawnSync } = require('child_process');
+const path = require('path');
+const { inspect } = require('util');
+
+const srcPath = path.resolve(__dirname, '..', '..', '..');
+const patchExportFnPath = path.resolve(__dirname, 'export_all_patches.py');
+const configPath = path.resolve(__dirname, '..', 'patches', 'config.json');
+
+// Re-export all the patches to check if there were changes.
+const proc = spawnSync('python', [patchExportFnPath, configPath, '--dry-run'], {
+  cwd: srcPath
+});
+
+// Fail if patch exporting returned 1, e.g dry run failed.
+if (proc.status === 1) {
+  console.log(proc.stderr.toString('utf8'));
+  process.exit(1);
+}
```

Comment:
can this be done by just `python export_all_patches --dry-run` from the package.json list? or is the cwd not stable for that script?

Reply:
@nornagon unfortunately no - i tried that initially to golf down changes by just putting `python script/export_all_patches.py patches/config.json`  in the lint-staged script and it's not stable:
```
fatal: cannot change to 'src': No such file or directory
fatal: cannot change to 'src': No such file or directory
Traceback (most recent call last):
File "script/export_all_patches.py", line 35, in <module>
  main()
File "script/export_all_patches.py", line 31, in main
  export_patches(json.load(config_json), dry_run)
File "script/export_all_patches.py", line 13, in export_patches
  git.export_patches(repo=repo, out_dir=patch_dir, dry_run=dry_run)
File "/Users/codebytere/Developer/electron-gn/src/electron/script/lib/git.py", line 265, in export_patches
  patch_range, num_patches = guess_base_commit(repo)
File "/Users/codebytere/Developer/electron-gn/src/electron/script/lib/git.py", line 184, in guess_base_commit
  return subprocess.check_output(args).rsplit('-', 2)[0:2]
File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 223, in check_output
  raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['git', '-C', u'src', 'describe', '--tags']' returned non-zero exit status 128
husky > pre-commit hook failed (add --no-verify to bypass)
```